script.js

```diff
@@ -1 +1,348 @@
-// Please implement exercise logic here
+/*#############
+GLOBAL VARIABLES
+##############*/
+// the element that contains the rows and squares
+let boardElement;
+// the element that contains the entire board
+// we can empty it out for convenience
+let boardContainer;
+// current player global starts at X
+const huPlayer = "X";
+const aiPlayer = "O";
+let currentPlayer = huPlayer;
+let gameMode = "AI";
+let gameInfo;
+let userGame;
+// keep data about the game in a 2-D array
+let boardSize;
+let board;
+let simulatedBoard;
+let lockBoard = false;
+let emptyCells = [];
+let blockMove = [];
+let winMove = [];
+let cornerMove = [];
+
+/*#############
+HELPER FUNCTIONS
+##############*/
+
+// initialize Matrix
+const Matrix =(n) => {
```

Comment:
I would like to make a few points here:

1. It is not incorrect to declare a function with capital letters, in fact - it is one of the many Javascript conventions called Constructor Invocation Pattern](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new). However, this is usually used when you are creating a new object and assigning some properties to it. In our use case, the function simply initializes the board and no properties are assigned to it (it is rather assigning a value to the `board` variable. Which takes us to the second point - 
2. The standard practice to declare a JavaScript function is to use [camelCasing](https://en.wikipedia.org/wiki/Camel_case). It is also better to use descriptive names, usually verbs in the imperative form (ie. initializeMatrix), so that if you are working in a team, your teammates can know what the function does at first glance (without having to read the comments).
3. I see that this function is only used once (in line 312), and we are using the returned value of this function. Usually, if we only need a function to carry some operation (ie. assign a value to `board`), then it is okay to not return anything. (Unless you got some editor warning / error from this)